### PR TITLE
feat: create CW proxy endpoint

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#228

### Dependencies

- Upstream: none
- Downstream: none

### Description

Commonwell make FHIR-like requests, not 100% compatible with the spec, so we're creating a proxy on our API so we can transform the request before forwarding it to our FHIR server.

This PR:
- adds the proxy endpoint on our API server
- adds the API GW resource/endpoint to access that proxy endpoint
- adds the Cognito user pool based on OAuth 2 client secret (used by CW to obtain the JWT and by API GW to validate said token)

There should be a follow-up PR to:
- create the infra and CI/CD for the FHIR server
- update the proxy endpoint created here to use the internal address of the FHIR server (currently, this is fixed to an external subdomain on R53)

### Release Plan

- no breaking changes, anytime

### CODE REVIEW

File: api/app/docker-compose.dev.yml [hunk 1]
```diff
@@ -33,6 +33,8 @@ services:
       - "9229:9229"
     volumes:
       - ./src:/usr/src/app/src
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (leite08) [inline api/app/docker-compose.dev.yml:37]: Allows us to use `host.docker.internal` to access the host's IP and, thus, other apps running local

File: api/app/package.json [hunk 2]
```diff
@@ -27,6 +27,7 @@
     "dayjs": "^1.11.6",
     "dotenv": "^16.0.3",
     "express": "^4.18.1",
+    "express-http-proxy": "^1.6.3",
```
Comment (leite08) [inline api/app/package.json:30]: I tried to use https://www.npmjs.com/package/http-proxy and https://www.npmjs.com/package/http-proxy-middleware with no success, so went with the next one in popularity that fit our use case.

File: api/app/src/routes/fhir-proxy.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import proxy from "express-http-proxy";
+
+// TODO make this dynamic/config/secret
+const router = proxy("https://fhir.staging.metriport.com", {
```
Comment (leite08) [inline api/app/src/routes/fhir-proxy.ts:4]: To be made dynamic as part of another PR, there's an item for this on the ticket's TODO: https://github.com/metriport/metriport-internal/issues/228

File: infra/lib/api-stack.ts [hunk 4]
```diff
@@ -318,53 +321,15 @@ export class APIStack extends Stack {
...
-      integrationHttpMethod: "ANY",
-      uri: `http://${fargateService.loadBalancer.loadBalancerDnsName}/{proxy}`,
-    });
-    tokenProxy.addMethod("ANY", integrationToken, {
-      requestParameters: {
-        "method.request.path.proxy": true,
-      },
-      authorizer: tokenAuth,
-    });
+    this.setupAPIGWApiTokenResource(id, api, link, tokenAuth, apiServerAddress);
```
Comment (leite08) [inline infra/lib/api-stack.ts:327]: Just moved this to its own function to move towards a more maintainable main function. Can provide code comparison if needed.

File: infra/lib/api-stack.ts [hunk 5]
```diff
@@ -496,6 +465,155 @@ export class APIStack extends Stack {
...
+
+  private setupOAuthUserPool(): cognito.IUserPool {
+    const userPool = new cognito.UserPool(this, "oauth-client-secret-user-pool", {
+      accountRecovery: cognito.AccountRecovery.EMAIL_ONLY,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+    // TODO make this a custom domain
+    userPool.addDomain("metriport-cognito-domain", {
+      cognitoDomain: {
+        domainPrefix: "metriport", // TODO make this dynamic/config
```
Comment (leite08) [inline infra/lib/api-stack.ts:536]: This should be updated when we create a custom domain for cognito, as part of the ticket (there's a TODO for it): https://github.com/metriport/metriport-internal/issues/228

**Review by Goncharo — APPROVED** [review]
Aside from the TODOs - 🚢🚢🚢
